extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.redis.client.deployment;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.Feature;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
+import io.vertx.redis.client.impl.types.BulkType;
+
+public class RedisClientProcessor {
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(Feature.REDIS_CLIENT);
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(Feature.REDIS_CLIENT.getName());
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem registerAPIsProducer() {
+        return AdditionalBeanBuildItem.unremovableOf("io.quarkus.redis.client.runtime.RedisAPIProducer");
```

Comment:
Why?

Reply:
We had some issues with beans not being injectable in gRPC for instance.

I might admit that I have no idea if it will be the case here. @mkouba could you confirm if we need the synthetic bean approach or if the producer approach will fly here?

Reply:
Synthetic beans are the way to go if there is a possibility that someone attempts to inject `@Dependent/@Singleton` bean produced by an uninitialized producer bean (usually before the `StartupEvent` is fired). From what I see the `RedisAPIProducer` is not initialized by a recorder so I think it's OK.